crates/holochain/src/core/ribosome/real_ribosome.rs

```diff
@@ -162,24 +171,58 @@ impl RealRibosome {
                         }
                         Err(wasm_error) => Err::<_, WasmError>(wasm_error),
                     };
-                    $crate::holochain_wasmer_host::import::set_context_data(ctx, result)
+                    env.set_data(result)
+                        .map_err(|e| RuntimeError::new(e.to_string()))?;
+                    Ok(vec![])
                 }
             }};
         }
         let mut imports = imports! {};
-        let mut ns = Namespace::new();
+        let mut ns = Exports::new();
+        let signature = FunctionType::new(vec![Type::I32, Type::I32], vec![]);
 
         // standard memory handling used by the holochain_wasmer guest and host macros
         ns.insert(
             "__import_data",
-            func!(holochain_wasmer_host::import::__import_data),
+            Function::new_native_with_env(
+                store,
+                env.clone(),
+                holochain_wasmer_host::import::__import_data,
+            ),
         );
 
         // imported host functions for core
-        ns.insert("__trace", func!(invoke_host_function!(trace)));
-        ns.insert("__hash_entry", func!(invoke_host_function!(hash_entry)));
-        ns.insert("__version", func!(invoke_host_function!(version)));
-        ns.insert("__unreachable", func!(invoke_host_function!(unreachable)));
+        ns.insert(
+            "__trace",
+            Function::new_with_env(store, &signature, env.clone(), invoke_host_function!(trace)),
+        );
+        ns.insert(
+            "__hash_entry",
+            Function::new_with_env(
+                store,
+                &signature,
+                env.clone(),
+                invoke_host_function!(hash_entry),
```

Comment:
Would be nice if there were a closure defined in this function to collapse this back down to a single line, just to remove boilerplate